clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -377,14 +392,9 @@ def _is_node_safe_to_kill(self, node_ip: str) -> bool:
         return True
 
     def _get_nodes_by_ip(self) -> Mapping[str, KubernetesNode]:
-        pool_label_selector = self.pool_config.read_string("pool_label_key", default="clusterman.com/pool")
-        pool_nodes = self._core_api.list_node().items
-
-        return {
-            get_node_ip(node): node
-            for node in pool_nodes
-            if not self.pool or node.metadata.labels.get(pool_label_selector, None) == self.pool
-        }
+        kwargs = {"label_selector": ",".join(self._label_selectors)} if self._label_selectors else {}
+        pool_nodes = self._core_api.list_node(**kwargs).items
+        return {get_node_ip(node): node for node in pool_nodes}
```

Comment:
nice!

Reply:
Thanks! I've see this was already in the works for #177, but I guess that's not getting shipped soon 😢